pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -120,9 +121,52 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 	return can, err
 }
 
-// admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
+// admitBlobWrite checks whether the blob does not exceed image quota or limit ranges, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
+		return err
+	}
+
+	if err := admitQuotas(ctx, repo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
```

Comment:
Well, the `OPENSHIFT_ENFORCE_QUOTA` is needed anyway to pass the value of `--disable-quota-limits` or `--enforce-quota-limits` to the registry. I'm concerned about existing deployments as well. That's why I'd rather default to `OPENSHIFT_ENFORCE_QUOTA=0` in the registry itself (if not specified in the dc). We should, however, default to `1` for any new deployments.